packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx

```diff
@@ -0,0 +1,369 @@
+import { useApolloClient, useQuery } from "@apollo/client";
+import {
+    BreadcrumbItem,
+    createOffsetLimitPagingAction,
+    EditDialog,
+    IFilterApi,
+    ISelectionApi,
+    PrettyBytes,
+    useStoredState,
+    useTableQuery,
+    useTableQueryPaging,
+} from "@comet/admin";
+import { DataGrid } from "@mui/x-data-grid";
+import * as React from "react";
+import { FileRejection, useDropzone } from "react-dropzone";
+import { FormattedDate, FormattedMessage, FormattedTime, useIntl } from "react-intl";
+import { useDebouncedCallback, useThrottledCallback } from "use-debounce";
+
+import { GQLDamFolderQuery, GQLDamFolderQueryVariables, GQLDamItemsListQuery, GQLDamItemsListQueryVariables } from "../../graphql.generated";
+import { useDamAcceptedMimeTypes } from "../config/useDamAcceptedMimeTypes";
+import { DamConfig, DamFilter } from "../DamTable";
+import AddFolder from "../FolderForm/AddFolder";
+import EditFolder from "../FolderForm/EditFolder";
+import { clearDamItemCache } from "../helpers/clearDamItemCache";
+import { isFile } from "../helpers/isFile";
+import { isFolder } from "../helpers/isFolder";
+import DamContextMenu from "./DamContextMenu";
+import { useFileUpload } from "./fileUpload/useFileUpload";
+import { damFolderQuery, damItemsListQuery } from "./FolderDataGrid.gql";
+import * as sc from "./FolderDataGrid.sc";
+import { DamFooter } from "./footer/DamFooter";
+import { DamItemLabelColumn } from "./label/DamItemLabelColumn";
+import { TableHead } from "./TableHead";
+import { useDamSearchHighlighting } from "./useDamSearchHighlighting";
+
+export type FooterType = "upload" | "selection";
+interface FooterInfo {
+    type: FooterType;
+
+    folderName?: string;
+}
+
+export type DamItemSelectionMap = Map<string, "file" | "folder">;
+
+interface FolderDataGridProps extends DamConfig {
+    id?: string;
+    breadcrumbs?: BreadcrumbItem[];
+    filterApi: IFilterApi<DamFilter>;
+    selectionApi: ISelectionApi;
+}
+
+const FolderDataGrid = ({
+    id,
+    filterApi,
+    breadcrumbs,
+    selectionApi,
+    hideContextMenu,
+    hideArchiveFilter,
+    hideMultiselect,
+    renderDamLabel,
+    TableContainer: ConfigTableContainer,
+    ...props
+}: FolderDataGridProps): React.ReactElement => {
+    const intl = useIntl();
+    const apolloClient = useApolloClient();
+
+    const [selectionMap, setSelectionMap] = React.useState<DamItemSelectionMap>(new Map());
+    const [footerInfo, setFooterInfo] = React.useState<FooterInfo | null>(null);
```

Comment:
The showing/hiding of the footer is very complicated. This is presumably because of the selection and upload type. Wouldn't it be easier to use two different instances of the footer and render them accordingly?

Reply:
Yes it's a bit complicated. The idea was to only have one state and one instance of the footer to ensure that the two footers are not rendered simultaneously.

I now split it up into to instances and states. That also works well.